### PR TITLE
[INLONG-2383][SDK] Support Kafka to consume PB compressed message protocol

### LINKED ISSUE #2383: [Feature][SDK] Sort-sdk support Kafka to consume PB compressed message protocol.
### Description

 Sort-sdk support Kafka consumer of PB compression cache message protocol.
Parent issue: #2375 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wardlican**
### Title Name: [INLONG-2383][SDK] Support Kafka to consume PB compressed message protocol

Fixes #2383

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [X] Make sure that the change passes the CI checks.

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
+        }
+        logger.info("closed {}", inLongTopic);
+        return true;
+    }
+
+    @Override
+    public boolean isClosed() {
+        return false;
```
Comment (vernedeng) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java:126]: why return false always?

**Review by luchunliang — APPROVED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+                context.getStatManager()
+                        .getStatistics(context.getConfig().getSortTaskId(),
+                                inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                        .addCallbackTimeCost(System.currentTimeMillis() - start).addCallbackDoneTimes(1);
+            } catch (Exception e) {
+                context.getStatManager()
+                        .getStatistics(context.getConfig().getSortTaskId(),
+                                inLongTopic.getInLongCluster().getClusterId(), inLongTopic.getTopic())
+                        .addCallbackErrorTimes(1);
+                e.printStackTrace();
```
Comment (vernedeng) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java:225]: should print log

**Review by vernedeng — APPROVED** [review]

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java (hunk 1 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java:126]: not complete yet

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java (hunk 2 above)
Comment (wardlican) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/InLongKafkaFetcherImpl.java:225]: ok

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**